src/shared/string-avatar.ts

```diff
@@ -1,27 +1,28 @@
-function stringToColor(string: string) {
-	let hash = 0;
-	let i;
+import { UserService } from 'clients/CoreService';
+import { useMemo } from 'react';
+import { useQuery } from 'react-query';
 
-	/* eslint-disable no-bitwise */
-	for (i = 0; i < string.length; i += 1) {
-		hash = string.charCodeAt(i) + ((hash << 5) - hash);
-	}
+/**
+ * Upload file to S3 with previously received pre-signed POST data.
+ * @param postUrl
+ * @param file
+ * @returns {Promise<void>}
+ */
+export const GetUserIconUrl = (id: number): string => {
+	const { data: content, isLoading: isContentLoading } = useQuery('users/icon', () =>
```

Comment:
You can't call hooks in shared functions. You can call hooks only in components(That's why this function is written in capital letter, but it doesn't mean that this function is a component) or hooks. Create new component `Avatar` instead, it will take userId as a parameter and you can throw `{...rest}` parameters where you can set ref and onClick. Than, you will be able to reuse this component.